### PR TITLE
feat(tile): form control

### PR DESCRIPTION
**Pull request by bennypowers**
## What I did

1. make tile a FACE

## Testing Instructions

1. check the unit tests, is anything missing?
2. manual testing in all the tile demos

### CODE REVIEW

**Comment by nikkimk** [comment]
@bennypowers there's a visual regression.
Screenshot 2023-10-25 at 10 49 34 AM
Screenshot 2023-10-25 at 10 49 47 AM

**Review by nikkimk — CHANGES_REQUESTED** [review]
There's a visual regression. Also RTI is now broken on tile groups.

**Comment by bennypowers** [comment]
the visreg is probably because i replaced things like `<div id="foo"><slot name="foo">` with `<slot id="foo" name="foo">`. probably i missed one of them in the shadow css file, or maybe they need `display: inline` instead of `display: block`, or some such. 

if you have time Nikki after demos PR would you mind taking a look? otherwise i'll give it a peek tomorrow

**Comment by nikkimk** [comment]
> the visreg is probably because i replaced things like `<div id="foo"><slot name="foo">` with `<slot id="foo" name="foo">`. probably i missed one of them in the shadow css file, or maybe they need `display: inline` instead of `display: block`, or some such.
> 
> if you have time Nikki after demos PR would you mind taking a look? otherwise i'll give it a peek tomorrow

@bennypowers Yeah. Those divs were necessary to get the layout needed. I tried to do it in fewer when I first set it up. I can do it but it will most likely entail putting them back in.

**Comment by bennypowers** [comment]
Slot has display: contents by default which means that we far as CSS is concerned, is as if the shed content is directly there in the div

So if you remove the div and apply all it's properties to the slot (including default ones) it should just work

The exception would be if you're already setting a display property on the slot

File: elements/rh-tile/rh-tile.ts [hunk 1]
```diff
@@ -141,6 +141,7 @@ export class RhTile extends LitElement {
   override async willUpdate(changed: PropertyValues<this>) {
     this.#internals.role = this.radioGroup ? 'radio' : this.checkable ? 'checkbox' : null;
     this.#internals.ariaChecked = !this.#isCheckable ? null : String(!!this.checked);
+    this.#internals.ariaLabel = !this.#isCheckable ? null : 'headline';
```
Comment (bennypowers) [inline elements/rh-tile/rh-tile.ts:144]: see test output. this line causes the tile to announce as 'headline', whereas previously it announced as 'Tile 1'. I believe the test case below is the correct behaviour, meaning that the element's text content should be the label. I will revert this line but preserve the rest of the fixes from this commit, and please review: am i seeing this clearly?

image

Comment (bennypowers) [reply elements/rh-tile/rh-tile.ts]: @nikkimk see https://github.com/RedHat-UX/red-hat-design-system/pull/1302/commits/6e68ff0de350510a25b8b3f7cdc0f15f234854eb

Comment (nikkimk) [reply elements/rh-tile/rh-tile.ts]: @bennypowers according to our Guidelines in docs, the heading slot is what should be used: https://ux.redhat.com/elements/tile/guidelines/#heading

Comment (bennypowers) [reply elements/rh-tile/rh-tile.ts]: Ok, we can adjust the code which computes the label to only consider elements slotted into the headline slot. 

but what if there's no headline text, we still need the accessible-label escape hatch, right?

Comment (nikkimk) [reply elements/rh-tile/rh-tile.ts]: Let's confirm with @marionnegp, but if I'm not mistaken selectable tiles should always have a headline. 

https://xd.adobe.com/view/925ed24c-3ddf-4111-bacf-2a42e5249f7a-c70d/screen/16207897-40cc-4e49-b064-2940e769b11b

Comment (marionnegp) [reply elements/rh-tile/rh-tile.ts]: Yes, the headline should be the text for the option you're selecting.

Comment (bennypowers) [reply elements/rh-tile/rh-tile.ts]: It's it possible to have a selectable tile without a headline?

Comment (marionnegp) [reply elements/rh-tile/rh-tile.ts]: Selectable tiles aren't supposed to have images or icons, so they should always have a text headline.

**Comment by bennypowers** [comment]
What's left to do here? I lost track...

**Comment by markcaron** [comment]
@nikkimk can you advise on where we're at?

File: elements/rh-tile/test/rh-tile-group.spec.ts [hunk 2]
```diff
@@ -72,84 +33,181 @@ describe('<rh-tile-group>', function() {
     });
   });
 
-  describe('radio tile', function() {
+  describe('as a radio group', async function() {
     beforeEach(async function() {
-      element = await createFixture<RhTileGroup>(tileGroup);
+      element = await createFixture<RhTileGroup>(html`
+        <rh-tile-group radio>
+          <rh-tile>Tile 1</rh-tile>
```
Comment (nikkimk) [inline elements/rh-tile/test/rh-tile-group.spec.ts:40]: Tiles should always have a headline.

```
<rh-tile><h1 slot="headline"><a href="#">Tile 1</a></h1></rh-tile>
<rh-tile><h1 slot="headline"><a href="#">Tile 2</a></h1></rh-tile>
<rh-tile><h1 slot="headline"><a href="#">Tile 3</a></h1></rh-tile>
```

**Review by nikkimk — CHANGES_REQUESTED** [review]

File: elements/rh-tile/test/rh-tile-group.spec.ts (hunk 2 above)
Comment (nikkimk) [reply elements/rh-tile/test/rh-tile-group.spec.ts:40]: @bennypowers I see you made changes. lmk when you're ready for me to tal

**Review by nikkimk — APPROVED** [review]
Livestock grazing tomorrow’s manure

**Pull request merged**